db/db_impl.cc

```diff
@@ -2737,6 +2738,78 @@ Status DBImpl::IngestExternalFile(
   return status;
 }
 
+Status DBImpl::VerifyChecksum() {
+  Status s;
+  std::vector<ColumnFamilyData*> cfd_list;
+  {
+    InstrumentedMutexLock l(&mutex_);
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
+      if (!cfd->IsDropped() && cfd->initialized()) {
+        cfd->Ref();
+        cfd_list.push_back(cfd);
+      }
+    }
+  }
+  for (auto cfd : cfd_list) {
+    VersionStorageInfo* vstorage = cfd->current()->storage_info();
+    for (int i = 0; i < vstorage->num_non_empty_levels() && s.ok(); i++) {
```

Comment:
how do you prevent files from being deleted as this runs on a live db? can you use `GetReferencedSuperVersion` / `ReturnThreadLocalSuperVersion`?